staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go

```diff
@@ -101,6 +101,11 @@ func WithLateConnectionFilter(handler http.Handler) http.Handler {
 func WithNonReadyRequestLogging(handler http.Handler, hasBeenReadyCh <-chan struct{}) http.Handler {
 	var nonReadyRequestReceived atomic.Bool
 
+	if hasBeenReadyCh == nil {
+		klog.Error("WithNonReadyRequestLogging: hasBeenReadyCh is nil, not setting up the handler")
```

Comment:
how about `utilruntime.HandleError` ?

Reply:
would drop the whole error. The return is enough. But also remove the `if hasBeenReadyCh != nil {` some lines down.
`